Samples/src/DatabaseSearch/DatabaseSearch.qs

```diff
@@ -304,31 +305,30 @@ namespace Microsoft.Quantum.Samples.DatabaseSearch {
     operation GroverHardCodedTest () : Unit {
 
         for (nDatabaseQubits in 0 .. 4) {
-
             for (nIterations in 0 .. 5) {
-
                 using ((markedQubit, databaseRegister) = (Qubit(), Qubit[nDatabaseQubits])) {
-                    QuantumSearch(nIterations, markedQubit, databaseRegister);
-                    let successAmplitude = Sin(IntAsDouble(2 * nIterations + 1) * ArcSin(1.0 / Sqrt(IntAsDouble(2 ^ nDatabaseQubits))));
-                    let successProbability = successAmplitude * successAmplitude;
+                    SearchForMarkedState(nIterations, markedQubit, databaseRegister);
+                    let dimension = IntAsDouble(2 ^ nDatabaseQubits);
+                    let successAmplitude = Sin(
+                        IntAsDouble(2 * nIterations + 1) *
+                        ArcSin(1.0 / Sqrt(dimension))
+                    );
+                    let successProbability = PowD(successAmplitude, 2.0);
                     AssertProb([PauliZ], [markedQubit], One, successProbability, "Error: Success probability does not match theory", 1E-10);
 
                     // If this result is One, we have found the marked
                     // element.
-                    let result = M(markedQubit);
+                    let isMarked = MResetZ(markedQubit) == One;
```

Comment:
I think the direct use of `== One` reads a little nicer here, but I like the suggestion of showing off `IsResultOne` as well. Teleportation seems like a good place to do so, so that we can show what each way of comparing to `One` looks like and comment on where you might use each.